frontend/src/views/WheresMyGene/components/HeatMap/components/XAxisChart/index.tsx

```diff
@@ -66,14 +75,41 @@ function GeneButton({
           active={genesToDelete.includes(geneName)}
           font={currentFont}
         >
+          <InfoButtonWrapper
+            data-test-id="gene-info-button-x-axis"
+            className={EXCLUDE_IN_SCREENSHOT_CLASS_NAME}
+            onClick={() => {
+              generateGeneInfo(geneName);
+
+              track(EVENTS.WMG_GENE_INFO, {
+                gene: geneName,
+              });
+
+              if (!dispatch) return;
```

Comment:
feels like this guard clause should be before `generateGeneInfo()`, because without dispatch sending analytics and generating gene info don't do anything anyway?